### PR TITLE
Declare ResponseEntity class in UserJWTController and UserResource

### LINKED ISSUE #6558: Always declare ResponseEntity class in controller methods
##### **Overview of the issue**

Now that Problem is used everywhere to handle errors we could avoid using an empty ResponseEntity in method declarations. Instead we could always use:
`ResponseEntity<Void>`
or 
`ResponseEntity<MyClass>`

Only two classes are affected:
**UserResource**
`public ResponseEntity createUser(@Valid @RequestBody ManagedUserVM managedUserVM) throws URISyntaxException`

**UserJWTController**
`public ResponseEntity authorize(@Valid @RequestBody LoginVM loginVM, HttpServletResponse response)`

##### **Motivation for or Use Case**

A method with this kind of declaration is not handled properly by swagger springfox, resulting in having an object like this:

```
"ResponseEntity": {
"type": "object",
"properties": {
"body": {
"type": "object"
},
"statusCode": {
"type": "string",
"enum": [
"100",
"101",
...
"statusCodeValue": {
"type": "integer",
"format": "int32"
}
```

In the _definitions_ of the generated swagger json file.

##### **Suggest a Fix**

Since  #6404 fixed the issue of having controllers handling both errors and responses. We could change the methods into something like this.

`public ResponseEntity<JWTToken> authorize(@Valid @RequestBody LoginVM loginVM, HttpServletResponse response)`
and
`public ResponseEntity<User> createUser(@Valid @RequestBody ManagedUserVM managedUserVM) throws URISyntaxException`

This would result in a cleaner swagger json.

##### **JHipster Version(s)**

4.10.0

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by encos**
Fix #6558

- Please make sure the below checklist is followed for Pull Requests.

- [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java [hunk 1]
```diff
@@ -60,23 +60,18 @@ public UserJWTController(TokenProvider tokenProvider, AuthenticationManager auth
 
     @PostMapping("/authenticate")
     @Timed
-    public ResponseEntity authorize(@Valid @RequestBody LoginVM loginVM, HttpServletResponse response) {
+    public ResponseEntity<JWTToken> authorize(@Valid @RequestBody LoginVM loginVM, HttpServletResponse response) {
```
Comment (cbornet) [inline generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java]: ResponseEntity wrapper can be removed

Comment (cbornet) [reply generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java]: Or maybe the response Authorization header could be set in the ResponseEntity and we can remove the HttpServletResponse.

File: generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java [hunk 2]
```diff
@@ -60,23 +60,18 @@ public UserJWTController(TokenProvider tokenProvider, AuthenticationManager auth
...
-            return new ResponseEntity<>(Collections.singletonMap("AuthenticationException",
-                ae.getLocalizedMessage()), HttpStatus.UNAUTHORIZED);
-        }
+        Authentication authentication = this.authenticationManager.authenticate(authenticationToken);
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        boolean rememberMe = (loginVM.isRememberMe() == null) ? false : loginVM.isRememberMe();
+        String jwt = tokenProvider.createToken(authentication, rememberMe);
+        response.addHeader(JWTConfigurer.AUTHORIZATION_HEADER, "Bearer " + jwt);
+        return ResponseEntity.ok(new JWTToken(jwt));
+
```
Comment (cbornet) [inline generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java]: Extra line

File: generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java (hunk 1 above)
Comment (encos) [reply generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java]: I'd keep it, that was the point of having the proper return type in swagger docs.
I'd go for removing the HttpServletResponse and returning something like this.
```
HttpHeaders httpHeaders = new HttpHeaders();
httpHeaders.add(JWTConfigurer.AUTHORIZATION_HEADER, "Bearer " + jwt);
return new ResponseEntity<>(new JWTToken(jwt), httpHeaders, HttpStatus.OK);
```

Comment (cbornet) [reply generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java]: Your swagger doc would be OK with a direct return of JWTToken. Anyway, :+1:  for ResponseEntity + auth header

**Comment by jdubois** [comment]
This is going to have merge conflicts because of #6563 (and it is a priority as the code doesn't even compile)

**Comment by encos** [comment]
Hi, don't know If I can help somehow. I just tried to rebase (locally) and there seem to be no merge conflicts. Probably I don't have the whole picture.

**Comment by jdubois** [comment]
Thanks @encos but in fact you're lucky, there's no conflict :-)

File: generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java [hunk 3]
```diff
@@ -60,23 +54,18 @@ public UserJWTController(TokenProvider tokenProvider, AuthenticationManager auth
...
-            log.trace("Authentication exception trace: {}", ae);
-            return new ResponseEntity<>(Collections.singletonMap("AuthenticationException",
-                ae.getLocalizedMessage()), HttpStatus.UNAUTHORIZED);
-        }
+        Authentication authentication = this.authenticationManager.authenticate(authenticationToken);
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+        boolean rememberMe = (loginVM.isRememberMe() == null) ? false : loginVM.isRememberMe();
+        String jwt = tokenProvider.createToken(authentication, rememberMe);
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.add(JWTConfigurer.AUTHORIZATION_HEADER, "Bearer " + jwt);
```
Comment (cbornet) [inline generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java]: I wonder if you could check the header's presence in the unit test ?

Comment (encos) [reply generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java]: Ok no problem

**Comment by jdubois** [comment]
If the test pass, let's merge this!

File: generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java [hunk 4]
```diff
@@ -47,8 +43,6 @@
 @RequestMapping("/api")
 public class UserJWTController {
 
-    private final Logger log = LoggerFactory.getLogger(UserJWTController.class);
```
[the remark this answers is not in the document]
Comment (cbornet) [reply generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java]: Isn't it going to make Sonar scream ?

[the remark this answers is not in the document]
Comment (cbornet) [reply generators/server/templates/src/main/java/package/web/rest/_UserJWTController.java]: Oh, I didn't see your other comment...

**Comment by jdubois** [comment]
@encos congrats on your first PR to the project!!!

**Comment by encos** [comment]
Thank you for the support and your jhi-efforts :)

**Pull request merged**